src/app.ts

```diff
@@ -0,0 +1,77 @@
+import dotenv from 'dotenv';
+import express from 'express';
+import morgan from 'morgan';
+import { join } from 'path';
+import { Writable } from 'stream';
+import { SpreadSheet } from '../spreadsheets/spread-sheet';
+import { spreadSheetUpdater } from '../spreadsheets/spread-sheet-updater';
+
+if (process.env.NODE_ENV === 'local') {
+  try {
+    dotenv.config({ path: join(__dirname, '..', '..', '.env') });
+  } catch (e) {
+    console.warn('Env file not loaded', e);
+  }
+}
+
+const sheetConfig = {
+  sheetId: process.env.sheetId,
+  updateIntervalTime: parseInt(process.env.updateIntervalTime, 10)
+};
+
+/**
+ * Used to perform Google API's authentication.
+ * You can create this object in the [google developers console](https://console.developers.google.com).
+ */
+const authData = {
+  auth_provider_x509_cert_url: process.env.auth_provider_x509_cert_url,
+  auth_uri: process.env.auth_uri,
+  client_email: process.env.client_email,
+  client_id: process.env.client_id,
+  client_x509_cert_url: process.env.client_x509_cert_url,
+  private_key: Buffer.from(process.env.private_key, 'base64').toString(),
+  private_key_id: process.env.private_key_id,
+  project_id: process.env.project_id,
+  token_uri: process.env.token_uri,
+  type: process.env.type
+};
+
+SpreadSheet.initSpreadSheet(sheetConfig.sheetId, authData); // initialize the spreadsheet
+spreadSheetUpdater(sheetConfig.updateIntervalTime); // continuously checks for updates in the spreadsheet after the specified time interval
+
+const app = express();
+
+// set up logging
+app.use((req, res, next) => {
```

Comment:
Why not just use vanilla morgan logging ? What is this "myStream" middleware doing that morgan doesn't give you ?

Reply:
I have currently changed it to use vanilla morgan logging.
But we can use the "myStream" option to pass in a configurable place where we want to do the logging. I had simply copied this from another place; where it was earlier configured to log to Kibana, but later removed and was simply logging to the console.